src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,171 @@
+package org.pesmypetcare.webservice.dao;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import org.pesmypetcare.webservice.entity.MedicationEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+import org.pesmypetcare.webservice.firebaseservice.FirebaseFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class MedicationDaoImpl implements MedicationDao{
+    private final String DELFAIL_KEY;
+    private final String MEDICATION_DOES_NOT_EXIST_EXC;
+    private final String INVALID_MEDICATION_EXC;
+
+    Firestore db;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)